tools/extra/c++utils/option_parser.cpp

```diff
@@ -172,7 +172,7 @@ bool option_parser::parse_args(int argc, char* argv[], option_map & known_option
     optarray.push_back({nullptr,0,nullptr,0});
 
     int longidx;
-    char c;
+    int c;
```

Comment:
getopt returns an integer

on aarch64 comparision of char c with -1 is throwing a compile error